Themis/ViewModels/Assessment/AssessmentViewModel.swift

```diff
@@ -1,72 +1,28 @@
 import Foundation
 import SwiftUI
+import Combine
 
 class AssessmentViewModel: ObservableObject {
-    @Published var fileTree: [Node] = []
-    @Published var openFiles: [Node] = []
-    @Published var selectedFile: Node?
-    @Published var editorFontSize = CGFloat(14) // Default font size
-
-    init(files: [String: FileType]) {
-        let node = ArtemisAPI.initFileTreeStructure(files: files)
-        self.fileTree = node.children ?? []
-    }
-
-    func openFile(file: Node) {
-        if !openFiles.contains(where: { $0.path == file.path }) {
-            openFiles.append(file)
-            file.fetchCode()
+    @Published var submission: SubmissionForAssessment?
+
+    var dismissPublisher = PassthroughSubject<Bool, Never>()
+    private var dismissView = false {
+        didSet {
+            DispatchQueue.main.async {
+                self.dismissPublisher.send(self.dismissView)
+            }
         }
-        selectedFile = file
     }
 
-    func closeFile(file: Node) {
-        openFiles = openFiles.filter({ $0.path != file.path })
-        selectedFile = openFiles.first
-    }
-
-    func incrementFontSize() {
-        editorFontSize += 1
-    }
-
-    func decrementFontSize() {
-        if editorFontSize > 8 { editorFontSize -= 1 }
+    @MainActor
+    func initRandomSubmission(exerciseId: Int) async {
+        do {
+            self.submission = try await ArtemisAPI.getRandomSubmissionForAssessment(exerciseId: 5284)
+        } catch {
+            print(error)
+            self.dismissView = true
```

Comment:
It is kind of weird that you will jump into the Assessment View but immediately are getting kicked out of it if you did not find a Submission. I think there should be some kind of ProgressView on the "Assess random submission" Button and only if you found a valid one switch to the Assessment View

Reply:
Yes for sure. It's intended that the logic before checks if there are still submissions to be assessed. This should only be a security, if something goes wrong. Doing it only with this one request, seemed not like its doable in a nice way.

Reply:
Hm ok but is there a different REST Call in order to check if there is even a submission?

Reply:
Do you have any idea how to implement it with a conditional navigation based on the async loaded response? The only way I could think of was with an if statement and no navigation link at all. But that seemed to me like a not so nice solution.

Reply:
This is not really testet, but usually I do something like this,

![Screenshot 2022-11-27 at 19 08 10](https://user-images.githubusercontent.com/16856573/204152309-d1f66983-480d-4239-bbc8-008057b7c103.png)

Reply:
Ok, I will try and see how it works

Reply:
I don't quite get when and how this should invoke the navigationDestination?

Reply:
Don't get how you mean this should work. How would the navigationDestination view modifier look like?

Reply:
Ok finally got it haha. Thank you Tom :)

Reply:
Nice :)